backend/backend/core/tools.py

```diff
@@ -32,7 +33,13 @@ def __init__(self):
 
     async def ainvoke_tool_call_artifact(self, query):
         results = await self.ainvoke(
-            {"args": {'query': query}, "type": "tool_call", "id": "foo", "name": "tavily"})
+            {
```

Comment:
The 'ainvoke' method is called with a dictionary that has an 'id' key set to 'foo'. This could be a potential security issue if the tool with id 'foo' has access to sensitive information.